### PR TITLE
Consolidate coveralls services

### LINKED ISSUE #818: Consolidate two Coveralls reporting APIs
See : https://github.com/axondeepseg/axondeepseg/pull/792#issuecomment-2182895874

We're using a combination of two different coveralls api, and though it worked before it no longer does for the purpose of Github status checks

### PR DESCRIPTION
**Pull request by mathieuboudreau**
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [x] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer
- [x] I've consulted [ADS's internal developer documentation](https://github.com/neuropoly/axondeepseg/wiki/Guidelines-for-contribution) to ensure my contribution is in line with any relevant design decisions

## Description

Coveralls reporting was broken, this PR fixes it by switching to only using the Coveralls Github Actions way (instead of a combination of that + the coveralls pipy)

## Linked issues

Resolves #818

### CODE REVIEW

**Comment by mathieuboudreau** [comment]
@hermancollin I've finished debugging/refactoring the coveralls setup - it should be good to go now ! Have a quick look and merge if you're ok with it - then we can see how it works with PR #800

File: .github/workflows/run_tests.yaml [hunk 1]
```diff
@@ -93,25 +93,26 @@ jobs:
...
       - name: Test with pytest
-        env:
-          COVERALLS_FLAG_NAME: ${{ matrix.test-name }}
-          COVERALLS_PARALLEL: true
-          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
         run: |
-          pip install coveralls
-          pytest . -v --cov AxonDeepSeg/ --cov-report term-missing
-          coveralls --service=github
+          pytest ./test/ -v --cov AxonDeepSeg/ --cov-report=lcov
```
Comment (joshuacwnewton) [inline .github/workflows/run_tests.yaml:97]: Just to fully understand this change:

- `lcov` has been added: Coveralls GitHub Action [requires LCOV format](https://github.com/coverallsapp/github-action?tab=readme-ov-file#usage).
- `term-missing` has been removed: Why is this?

File: .github/workflows/run_tests.yaml [hunk 2]
```diff
@@ -93,25 +93,26 @@ jobs:
...
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          pip3 install --upgrade coveralls
-          coveralls --finish
+    - name: Coveralls Finished
+      uses: coverallsapp/github-action@v2
+      with:
+        github-token: ${{ secrets.github_token }}
+        parallel-finished: true
+        carryforward: "ubuntu-latest,macos-12,windows-latest"
```
Comment (joshuacwnewton) [inline .github/workflows/run_tests.yaml:118]: The flag names are unique, and the `carryforward` matches the `flag-name` defined earlier, which matches the [example usage](https://github.com/coverallsapp/github-action?tab=readme-ov-file#complete-parallel-job-example). LGTM!

**Review by joshuacwnewton — APPROVED** [review]
Seems like a straightforward change to me! I just had one question about the change to the `pytest` command, but other than that, it LGTM, so I'm pre-approving.

File: .github/workflows/run_tests.yaml (hunk 1 above)
Comment (mathieuboudreau) [reply .github/workflows/run_tests.yaml:97]: >term-missing has been removed: Why is this?

Ahh, that's just to show which lines are being missed in the CLI run log on github actions. It's not really a requirement either way, it doesn't impact the coverage report or the tests themselves, just a style choice. I can reintegrate it if you think other ppl might want it when looking at the GitHub actions runs

**Comment by mathieuboudreau** [comment]
Thanks @joshuacwnewton !

**Pull request merged**